### PR TITLE
Fixed undefined `_WIN32_WINNT` for generated files on Windows.

### PR DESCRIPTION
**Pull request by chhsia0**
When compiling a `.grpc.cc` file generated by `protoc.exe` with `grpc_cpp_plugin.exe`, the following error would show up:
```
Please compile grpc with _WIN32_WINNT of at least 0x600 (aka Windows Vista)
```
It seems that the generated file somehow includes `port_platform.h`, which checks `_WIN32_WINNT` before including `windows.h`, which includes `sdkddkver.h`, which in turn defines `_WIN32_WINNT`. This PR moves the inclusion of `windows.h` before checking `_WIN32_WINNT` in this header file.

This patch is the same as https://github.com/grpc/grpc/pull/14574, and might be the same error described in https://github.com/grpc/grpc/issues/14575.

### CODE REVIEW

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* User @chhsia0 isn't covered by a CLA. They will need to complete the form at https://identity.linuxfoundation.org/projects/cncf

_Regards,
CLA GitHub bot_

**Comment by chhsia0** [comment]
Signed the CLA.

**Comment by chhsia0** [comment]
My original summary was incorrect and it has been updated now.

**Comment by IronsDu** [comment]
Good.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by nicolasnoble** [comment]
So, well, your PR partially nullifies the `#if` statement that is right below: https://github.com/grpc/grpc/blob/master/include/grpc/impl/codegen/port_platform.h#L44

The point is, and the main design paradigm around gRPC is that we want to be as boring as possible, and follow the [principle of least surprise](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) as much as possible. Following this principle means that the user needs to be fully explicit about what is going on in their code, and this condition here is to force the user to somehow explicitly define the version of the Windows API they want to use without us forcing one by default and hence polluting their environment, creating a weird situation where parts of their code is compiled for one version, and parts of their other code is compiled for another version.

Long story short, this means the easiest way, and in my opinion the most correct way, is for you to always define `_WIN32_WINNT` from the compilation flags. Note that while this isn't necessarily the default from what Visual Studio does, but this is something that I personally consider to be bad practice, and [I'm not the only one with this stance](https://github.com/weidai11/cryptopp/issues/178#issuecomment-233405697).

This is the only way to ensure no discrepancy across the project is happening, thus ensuring the principle of least surprise I was talking about before. In fact, the way you got bitten by this tells me that you would otherwise have been silently be using two different versions of the WINNT API inside the same project, which is just wrong.

All of this being said, maybe the right thing to do would be to try and get the code generator to not end up including `port_platform.h`. I now wonder what causes this to happen.

**Comment by chhsia0** [comment]
My original intention to include `windows.h` earlier is that it keeps the user the choice to target a specific API version, but also could use the default API version that matches what is used in their project. So if a user want to target a specific API version, they will explicitly define `_WIN32_WINNT` for their own whole project; if a user only want to target the current API version in their environment, they'll leave it to the default.

But your concern also makes sense. It sounds better if we could avoid including this header in the generated C++ files. I'll investigate the possibility.

**Comment by chhsia0** [comment]
I spent some time expanding the head inclusion tree of the generated `.grpc.pb.cc` file and the results is attached, where a `+` sign means that the header will be included, a `-` sign means that the header will be skipped due to a previous inclusion, and a `*` sing is the inclusion of `port_platform.h`.

TL;DR: The main header files including `port_platform.h` are: `grpc/impl/codegen/byte_buffer.h`, `grpc/impl/codegen/core_codegen_interface.h`, `grpc/impl/codegen/propagation_bits.h`. It seems to me that there is no easy way to remove the inclusion. @nicolasnoble Thoughts?
[port_platform_dependency.txt](https://github.com/grpc/grpc/files/1973088/port_platform_dependency.txt)

**Comment by apolcyn** [comment]
Ack on relation to #14487 - if there's a problematic file for which it's beneficial to </i>not</i> include `port_platform.h`, maybe we could blacklist it from the [port platform inclusion sanity check script](https://github.com/grpc/grpc/blob/master/tools/run_tests/sanity/check_port_platform.py) ?

**Comment by yashykt** [comment]
@apolcyn @nicolasnoble let's get this done if still relevant
